core/application/commands.go

```diff
@@ -37,6 +37,29 @@ func (a *app) CreateMicroVM(ctx context.Context, mvm *models.MicroVM) (*models.M
 		return nil, fmt.Errorf("an error occurred when attempting to validate microvm spec: %w", validErr)
 	}
 
+	if mvm.Spec.RootVolume.Source.VirtioFS != nil {
```

Comment:
I would move this to a "custom" validator on the struct.  For example:

```go
	// RootVolume specified the root volume to be attached to the machine.
	RootVolume Volume `json:"root_volume" validate:"required,novirtiofs"`
```
([source](https://github.com/liquidmetal-dev/flintlock/blob/main/core/models/microvm.go#L41))

And then register the tag and add the validation function here https://github.com/liquidmetal-dev/flintlock/blob/main/pkg/validation/validate.go

Reply:
Can you check if the most recent code update resolves this?